opencensus/stats/internal/delta_producer.h

```diff
@@ -0,0 +1,121 @@
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef OPENCENSUS_STATS_INTERNAL_DELTA_PRODUCER_H_
+#define OPENCENSUS_STATS_INTERNAL_DELTA_PRODUCER_H_
+
+#include <cstdint>
+#include <memory>
+#include <thread>
+#include <unordered_map>
+#include <vector>
+
+#include "absl/synchronization/mutex.h"
+#include "absl/time/time.h"
+#include "opencensus/stats/bucket_boundaries.h"
+#include "opencensus/stats/distribution.h"
+#include "opencensus/stats/internal/measure_data.h"
+#include "opencensus/stats/measure.h"
+#include "opencensus/stats/tag_set.h"
+
+namespace opencensus {
+namespace stats {
+
+// Delta is thread-compatible.
```

Comment:
Reading data out of a Delta happens in a later PR?

Reply:
Correct.